hibernate-core/src/test/java/org/hibernate/orm/test/type/EnumArrayTest.java

```diff
@@ -77,6 +77,7 @@ public void startUp() {
 	}
 
 	@Test
+	@SkipForDialect(value = AltibaseDialect.class, comment = "Altibase can't distinguish between empty and null value")
```

Comment:
You mean an empty `varbyte` i.e. with length 0 and a `null` value are "the same" to Altibase? If so, then you should have a lot more test failures since all the `XArrayTest` classes in this package use the same sort of test data setup. Also, the dialect does not support SQL standard arrays, so the encoding will always fall back to VARBINARY encoding.

Reply:
In Altibase null and empty values cannot be distinguished. Could you please let me know test class names regarding this empty and null value issue? In fact, I didn't find a lot of this empty null related test failures.

Reply:
Pretty sure that all of the following array test classes insert a null `varbinary`:

- `BooleanArrayTest`
- `BasicListTest`
- `BasicSortedSetTest`
- `DateArrayTest`
- `DoubleArrayTest`
- `EnumArrayTest`
- `FloatArrayTest`
- `IntegerArrayTest`
- `LongArrayTest`
- `ShortArrayTest`
- `StringArrayTest`
- `TimeArrayTest`
- `TimestampArrayTest`
- `VarbinaryArrayTest`

Reply:
All these tests runs ok in Altibase.
I think javaType.unwrap() works diffrently depending on values.

In BooleanArrayTest
q.setParameter( "data", new Boolean[]{} );
==> st.setBytes( index, javaType.unwrap( value, byte[].class, options ) );
javaType.unwrap with Boolean object returns byte array with length 45

q.setParameter( "data", null );
==> st.setNull( index, st.getParameterMetaData().getParameterType( index ) ); 
null is called by PreparedStatement.setNull and stored as null in Altibase server

In EnumArrayTest
q.setParameter( "data", new MyEnum[]{} );
== >st.setBytes( index, javaType.unwrap( value, byte[].class, options ) );
javaType.unwrap with MyEnum object returns empty byte array and fetched as null in Altibase

Reply:
Ah ok, then don't worry about it :)